base_search_mail_content/models/mail_thread.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# © 2016-17 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# © 2016 Serpent Consulting Services Pvt. Ltd. (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from lxml import etree
+from odoo.osv import expression
+from odoo.osv.orm import setup_modifiers
+
+
+class MailThread(models.AbstractModel):
+
+    _inherit = 'mail.thread'
+
+    def _search_message_content(self, operator, value):
+
+        domain = [('model', '=', self._name)]
+        if operator not in expression.NEGATIVE_TERM_OPERATORS:
+            domain += ["|"] * 4
+        domain += [('record_name', operator, value),
+                   ('subject', operator, value),
+                   ('body', operator, value),
+                   ('email_from', operator, value),
+                   ('reply_to', operator, value)]
+        return domain
+
+    message_content = fields.Text(
+        string='Message Content',
+        help='Message content, to be used only in searches',
+        compute=lambda self: False,
+        search='_search_message_content')
+
+    @api.model
+    def _fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                         submenu=False):
+        """
+        Override to add message_ids field in all the objects
+        that inherits mail.thread
+        """
+        res = super(MailThread, self)._fields_view_get(
+            view_id=view_id, view_type=view_type, toolbar=toolbar,
+            submenu=submenu)
+        if view_type == 'search':
+            doc = etree.XML(res['arch'])
+            res['fields'].update({
+                'message_content': {
+                    'type': 'char',
+                    'string': 'Message content',
+                }
+            })
+
+            for node in doc.xpath("//field[1]"):
+                # Add message_content in search view
+                elem = etree.Element('field', {
+                    'name': 'message_content',
```

Comment:
This does not seem to be working, maybe yes it was required to inherit from `base` instead.